segmenters/nlp/SpacySentencizer/__init__.py

```diff
@@ -0,0 +1,96 @@
+from typing import Dict, List
+
+from jina.executors.segmenters import BaseSegmenter
+
+
+class SpacySentencizer(BaseSegmenter):
+    """
+    :class:`SpacySentencizer` split the text on the doc-level
+    into sentences on the chunk-level with spaCy as the backend.
+    The splitting technique is divided into two different approaches:
+    1) Default Segmenter: Utilizes dependency parsing to determine the rule for
+        splitting the text. For more info please refer to https://spacy.io/api/sentencizer.
+    2) Machine Learning-based Segmenter: Utilizes SentenceRecognizer model trained
+        on a dedicated data created for sentence segmentation. For more info please
+        refer to https://spacy.io/api/sentencerecognizer.
+
+    :param lang: pre-trained spaCy language pipeline
+    :param use_default_segmenter: boolean to determine which segmentation algorithm to use,
+        by default False.
+    :param args:  Additional positional arguments
+    :param kwargs: Additional keyword arguments
+    """
+
+    SPACY_COMPONENTS = [
+        'tagger',
+        'parser',
+        'ner',
+        'senter',
+        'entity_linker',
+        'textcat',
+        'entity_ruler',
+        'merge_noun_chunks',
+        'merge_entities',
+        'merge_subtokens',
+        'lemmatizer',
+        'attribute_ruler',
+    ]
+
+    def __init__(self, lang: str, use_default_segmenter: bool = False, *args, **kwargs):
+        """Set constructor."""
+        super().__init__(*args, **kwargs)
+        self.lang = lang
+        self.use_default_segmenter = use_default_segmenter
+
+    def post_init(self):
+        import spacy
+
+        try:
+            self.spacy_model = spacy.load(self.lang)
+            # Disable everything as we only requires certain pipelines to turned on.
+            ignored_components = []
+            for comp in self.SPACY_COMPONENTS:
+                try:
+                    self.spacy_model.disable_pipe(comp)
+                except Exception:
+                    ignored_components.append(comp)
+            self.logger.info(f'Ignoring {ignored_components} pipelines as it does not available on the model package.')
+        except IOError:
+            self.logger.error(
+                f'spaCy model for language {self.lang} can not be found. Please install by referring to the '
+                'official page https://spacy.io/usage/models.'
+            )
+            raise
+
+        if self.use_default_segmenter:
+            try:
+                self.spacy_model.enable_pipe('parser')
```

Comment:
This logic seems weird, why are u disabling the pipelines to later enable them?

Why is it so important to disable pipelines? Sorry I am not familiar with the tool

Reply:
I've explained my reasoning above
> spaCy works in a sequence of pipelines, so when a text enter spaCy it will be run through all of those pipes (pipes = pipelines). **I'm thinking to disable all the pipelines and just enable ones that we need to remove extra latency**

Why parser is disabled and enabled first because it depends on `use_default_segmenter`, right now the default value is False, which means will use the ml-based segmenter as the default. If we replace the default value of `use_default_segmenter` to True then we can remove this logic and exclude parser pipeline from being disabled in the beginning and disable it when the value changed to False. Would you prefer we do it the latter or the former?